### ISSUE TITLE
Open photos from a map

### ISSUE THREAD

**Issue by dbogdanov** [bpatrik/pigallery2#175]
Currently, when browsing a map one can preview the photos by clicking on them, but it is not possible to open those photos like when browsing an album.

**Comment by jezikk82** [comment]
It'd be nice to have option to open:
1. Album/folder that photo belong
2. All photos take in the same place/area (some small radius, ~ 500m-1km)

Moreover, please add option to settings or config.json to set default zoom on the map presented next to individual photo.

**Comment by bpatrik** [comment]
> All photos take in the same place/area (some small radius, ~ 500m-1km)

You can do such a search already `5-km-from:(New York) # photos 5 km from the center of New York `

**Comment by jezikk82** [comment]
Sorry but the search query for some strange reasons work only with bigger cities. When you put small town, village, different rural area I return 0 results. At the same time on the map I see that pictures.

**Comment by bpatrik** [comment]
That is odd. I'm using: https://github.com/nchaulet/node-geocoder#readme to get the gps coordinate for a village. 

It hapends here: https://github.com/bpatrik/pigallery2/blob/bae5559356e4d4320425415696e6fd6d24e5fc04/src/backend/model/database/LocationManager.ts#L14-L28

And it works for my little town (population < 3000) in Europe. 

There might be some issue with it close the to zero and max coordinates.

**Comment by jezikk82** [comment]
I checked over 20 small locations in 3 countries and found couple of places that are shown on the maps but can't be found. I checked in different gallery soft those pics and according to 'map box API' the address is different then the place on the map. I think it's some bug in the map provided. (geo encoding).

**Comment by jezikk82** [comment]
I'm also in Europe (Central part of it).

**Comment by bpatrik** [comment]
Can you give a not working example?

**Comment by jezikk82** [comment]
10-km-from:Ujazd
GPS from photo: 50.137772,19.819286
Xiaomi phone decode this cords as Ujazd 74, 32-080 Poland

**Comment by bpatrik** [comment]
To find a different city, try a more specific search.
`10-km-from:(Ujazd, Tupadły)`, uses:
```
[                                                                                                                      
  {                                                                                                                    
    latitude: 53.0030207,                                                                                              
    longitude: 17.5129951,                                                                                             
    formattedAddress: 'Ujazd, Tupadły, gmina Kcynia, powiat nakielski, województwo kujawsko-pomorskie, 89-240, Polska',
    country: 'Polska',                                                                                                 
    city: 'Tupadły',
    state: 'województwo kujawsko-pomorskie',
    zipcode: '89-240',
    streetName: undefined,
    streetNumber: undefined,
    countryCode: 'PL',
    neighbourhood: '',
    provider: 'openstreetmap'
  }
]

```

**Comment by mfaber** [comment]
This is indeed a feature that would greatly enhance the use of the map view. I often use the map to find specific photos. If found on the map, there's no easy way to see this (and the following or previous photos) in full resolution (only a large thumbnail available on the map). 

I'd like to suggest that clicking on the photo in the map view leads to the fullscreen view of the photo in the standard folder view mode (where you can then move forward or reverse through the photos as usual). 
Clicking the (X) would return the user to the map view again (position of the current photo in full view or the photo that was clicked on in map view, originally).

**Comment by yllekz** [comment]
I'd love this feature. I just moved over to this from Photoview and the one feature that this doesn't have is a general map that plots all the photos on it.

**Comment by karolleon** [comment]
Yes, please add this feature! The app is so close to fulfilling my dream of having a simple way to show my hikes and photos to friends and family. I would like to show them a map of GPX tracks (already possible) with photos, not just thumbnails.

**Comment by ericnotthered** [comment]
+1 

I would like to be able to 
- Select a single photo and go to it in the gallery
- Open a search for all of the photos shown in the current map view.  In other words, the photos in the current directory (and sub-directories, if a flattened view is selected) that are in the map rectangle currently being viewed.

**Comment by bpatrik** [comment]
The last commit should make the map popup images clickable. Clicking on them, brings you to the normal lightbox (preview) mode. 

It will take about 40 mins to build the next docker edge (since the FR is open for 5 years, I hope its ok to wait 40 more mins :D )

**Comment by mfaber** [comment]
Yay, thank you, @bpatrik !!! I just tried the edge version and it works very well!!

I am undecided whether one should return to the album gallery (current behavior) or the map upon clicking a photo in the map view and pressing (X) in the lightbox view.

**Comment by bpatrik** [comment]
With closing, we always navigate upwards (in hierarchy), You can use the back button to go back to the map.

**Comment by bpatrik** [comment]
Closing this as the feature is done.
